network-mux/src/Network/Mux.hs

```diff
@@ -72,6 +76,21 @@ data Mux (mode :: MuxMode) m =
        muxStatus          :: StrictTVar m MuxStatus
      }
 
+
+miniProtocolStateMap :: Mux mode m -> Map (MiniProtocolNum, MiniProtocolDir)
+                                          (StrictTVar m MiniProtocolStatus)
```

Comment:
This exposes the underlying TVars. Perhaps it should only return STM read actions.

Reply:
Yes, that would be better.